server/app/job.go

```diff
@@ -82,7 +82,8 @@ func (a *UserSurveyApp) startNewSurveyIfNeeded() error {
 
 		surveyFromConfig := &model.Survey{
 			ID:              utils.NewID(),
-			ExcludedTeamIDs: config.TeamFilter.FilteredTeamIDs,
+			FilterTeamIDs:   config.TeamFilter.FilteredTeamIDs,
+			TeamFilterType:  config.TeamFilter.FilterType,
```

Comment:
No changes to make here, but I wonder if this might've been easier to implement if this was represented internally as two arrays: `IncludedTeamIDs` and `ExcludedTeamIDs` since that would've been my first thought of how to store this. That seems like it might've made the code checking those arrays simpler, but it does mean that it's possible for this configuration to end up in a way that the UI can't represent

Reply:
The list teams never change on changing the options, the filter type just defines how the list of teams is to be treated. As such, its of no benefits to store them as two lists.

At a time, there can only be one behaviour - you include the list, or exclude, not both,. So no use of storing two lists.